modules/game_console/console.lua

```diff
@@ -177,25 +209,11 @@ function selectAll(consoleBuffer)
 end
 
 function toggleChat()
-  consoleToggleChat:setChecked(not consoleToggleChat:isChecked())
-end
-
-function updateChatMode()
-  if consoleToggleChat:isChecked() then
-    disableChat()
-  else
-    enableChat()
-  end
+  useAutoHideChat = consoleToggleChat:isChecked()
 end
```

Comment:
By modifying this method like this, you are breaking the hotkey configuration to change between wasd and chat mode in

https://github.com/mehah/otclient/blob/b3044107ce1991a736068dc24b6f847c57a284d7/modules/game_hotkeys/hotkeys_manager.lua#L442-L443

Reply:
i will update it more and propose changes again i still didnt finished it yet cause there are no initial states unfortunatly...